lib/pbench/server/api/resources/query_controllers.py

```diff
@@ -0,0 +1,273 @@
+from flask import request
+from flask_restful import Resource, abort
+import requests
+
+from dateutil import parser, rrule
+
+
+class QueryControllers(Resource):
+    """
+    Abstracted Pbench API to get date-bounded controller data.
+    """
+
+    def __init__(self, app, api):
+        self.app = app
+        self.api = api
+
+    @staticmethod
+    def _genMonthUriRange(prefix, index, start, end):
+        """
+        _genMonthUriRange Construct a comma-separated list of index names
```

Comment:
Since we are going to Elasticsearch V7, we should consider using the index life-cycle management, where we fill an index with data, and have it auto roll-over to another index name for the next one.

Then we just use a date range query across all the indices.  At least that is what the latest Kibana does.  We might want to consider this instead of building a long list of indices and handling the roll-over that way.

Reply:
A battle for another day.